packages/rke2-coredns/rke2-coredns.patch

```diff
@@ -314,24 +467,68 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/rke2-coredns/charts-original/templates
 +  {{ else }}
 +  clusterIP: {{ (lookup "v1" "ConfigMap" "kube-system" "cluster-dns").data.clusterDNS }}
    {{- end }}
-   {{- if .Values.service.externalTrafficPolicy }}
-   externalTrafficPolicy: {{ .Values.service.externalTrafficPolicy }}
+   {{- if .Values.service.externalIPs }}
+   externalIPs:
+diff -x '*.tgz' -x '*.lock' -uNr packages/rke2-coredns/charts-original/templates/service.yaml.orig packages/rke2-coredns/charts/templates/service.yaml.orig
+--- packages/rke2-coredns/charts-original/templates/service.yaml.orig
++++ packages/rke2-coredns/charts/templates/service.yaml.orig
+@@ -0,0 +1,42 @@
++apiVersion: v1
++kind: Service
++metadata:
++  name: {{ template "coredns.fullname" . }}
++  labels:
++    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
++    app.kubernetes.io/instance: {{ .Release.Name | quote }}
++    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
++    {{- if .Values.isClusterService }}
++    k8s-app: {{ .Chart.Name | quote }}
++    kubernetes.io/cluster-service: "true"
++    kubernetes.io/name: "CoreDNS"
++    {{- end }}
++    app.kubernetes.io/name: {{ template "coredns.name" . }}
++{{- if .Values.customLabels }}
++{{ toYaml .Values.customLabels | indent 4 }}
++{{- end }}
++  annotations:
++{{ toYaml .Values.service.annotations | indent 4 }}
++spec:
++  selector:
++    app.kubernetes.io/instance: {{ .Release.Name | quote }}
++    {{- if .Values.isClusterService }}
++    k8s-app: {{ .Chart.Name | quote }}
++    {{- end }}
++    app.kubernetes.io/name: {{ template "coredns.name" . }}
++  {{- if .Values.service.clusterIP }}
++  clusterIP: {{ .Values.service.clusterIP }}
++  {{- end }}
++  {{- if .Values.service.externalIPs }}
++  externalIPs:
++  {{ toYaml .Values.service.externalIPs | indent 4 }}
++  {{- end }}
++  {{- if .Values.service.externalTrafficPolicy }}
++  externalTrafficPolicy: {{ .Values.service.externalTrafficPolicy }}
++  {{- end }}
++  {{- if .Values.service.loadBalancerIP }}
++  loadBalancerIP: {{ .Values.service.loadBalancerIP }}
++  {{- end }}
++  ports:
++{{ include "coredns.servicePorts" . | indent 2 -}}
++  type: {{ default "ClusterIP" .Values.serviceType }}
 diff -x '*.tgz' -x '*.lock' -uNr packages/rke2-coredns/charts-original/values.yaml packages/rke2-coredns/charts/values.yaml
 --- packages/rke2-coredns/charts-original/values.yaml
 +++ packages/rke2-coredns/charts/values.yaml
-@@ -3,8 +3,8 @@
+@@ -3,7 +3,7 @@
  # Declare variables to be passed into your templates.
  
  image:
 -  repository: coredns/coredns
--  tag: "1.6.9"
 +  repository: rancher/hardened-coredns
-+  tag: "v1.6.9"
+   tag: "1.7.1"
    pullPolicy: IfNotPresent
  
- replicaCount: 1
-@@ -34,10 +34,10 @@
-     prometheus.io/port: "9153"
+@@ -66,10 +66,10 @@
+   annotations: {}
  
  serviceAccount:
```

Comment:
Do we want to flip this to true and include the name field like we currently have?

Reply:
I can't tell what this patch is actually doing - are we dropping the replicaCount, and Prometheus scrape port annotations?

Reply:
no, its just the change is no longer anchored to that text

Reply:
@briandowns create service account is being set to true via the values file